src/pybind/mgr/cephadm/inventory.py

```diff
@@ -163,12 +163,15 @@ def __init__(self, mgr):
         self.daemons = {}   # type: Dict[str, Dict[str, orchestrator.DaemonDescription]]
         self.last_daemon_update = {}   # type: Dict[str, datetime.datetime]
         self.devices = {}              # type: Dict[str, List[inventory.Device]]
+        self.osdspec_previews = {}     # type: Dict[str, List[Dict[str, Any]]]
         self.networks = {}             # type: Dict[str, Dict[str, List[str]]]
         self.last_device_update = {}   # type: Dict[str, datetime.datetime]
         self.daemon_refresh_queue = [] # type: List[str]
         self.device_refresh_queue = [] # type: List[str]
+        self.osdspec_previews_refresh_queue = [] # type: List[str]
         self.daemon_config_deps = {}   # type: Dict[str, Dict[str, Dict[str,Any]]]
         self.last_host_check = {}      # type: Dict[str, datetime.datetime]
+        self.loading_osdspec_preview = {}  # type: Dict[str, bool]
```

Comment:
at which point do we need to split-up the HostCache?

Reply:
split up in separate modules/classes 
* DaemonCache()
* HostCache()
* ...

you mean?

Reply:
as a followup then?

Reply:
not sure. I think it's ok to add those members here, but this is maybe somethign for the future.